src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerInEstablishmentResource.kt

```diff
@@ -0,0 +1,82 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.resource
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.Parameter
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.security.SecurityRequirement
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.data.domain.Page
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.prisonersearch.model.Prisoner
+import uk.gov.justice.digital.hmpps.prisonersearch.resource.advice.ErrorResponse
+import uk.gov.justice.digital.hmpps.prisonersearch.services.PrisonerInEstablishmentService
+import uk.gov.justice.digital.hmpps.prisonersearch.services.dto.PrisonerInEstablishmentRequest
+import javax.validation.Valid
+
+@RestController
+@Validated
+@PreAuthorize("hasAnyRole('ROLE_GLOBAL_SEARCH', 'ROLE_PRISONER_SEARCH')")
+class PrisonerInEstablishmentResource(private val searchService: PrisonerInEstablishmentService) {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  // A hack to allow swagger to determine the response schema which includes generic content
+  abstract class Response : Page<Prisoner>
```

Comment:
I don't remember having to do this hack to get swagger to do paging properly, will have a look to see how I've done it before

Reply:
Yes, agreed - copy and paste from a resource in true same project. I will double check why it was there